File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.js
function solution(numbers) {
    const result = new Array(numbers.length).fill(-1);
    
    for(let i = numbers.length-2; i >= 0; i--) {
        for(let j = i+1; j < numbers.length; j++) {
            if (numbers[i] < numbers[j]) {
                result[i] = numbers[j];
                break;
            } else {
                if (result[j] === -1) {
                    result[i] = -1;
                    break;
                } else if (numbers[i] < result[j]) {
                    result[i] = result[j];
                    break;
                }
            }
        }
    }
    
    return result;
}

File: 프로그래머스/2/154539. 뒤에 있는 큰 수 찾기/뒤에 있는 큰 수 찾기.py
def solution(numbers):
    result = [-1] * len(numbers)
    stack = []
    
    for i, num in enumerate(numbers):
        while stack and numbers[stack[-1]] < num:
            index = stack.pop()
            result[index] = num
        stack.append(i)
    
    return result
            
    
    